packages/core/core/src/types.ts

```diff
@@ -522,9 +522,11 @@ export interface AgnosticCoupon {
 export interface AgnosticDiscount {
   id: string;
   name: string;
-  description: string;
   value: number;
-  code?: string;
+  valueType: 'relative' | 'absolute' | 'fixed' | 'giftLineItem';
```

Comment:
I dont think these valueTypes are agnostic. Maybe we could add just a string here and use enum in the integration (CT)?

Reply:
Like [this](https://github.com/vuestorefront/vue-storefront/pull/6149/commits/67a560c93562346fb4c54cf89349528338747bff)? :)